### ISSUE TITLE
System telemetry warning about low voltage on Expansion Hub is only refreshed during a Restart Robot.

### ISSUE THREAD

**Issue by FROGbots-4634** [FIRST-Tech-Challenge/SkyStone#6]
The title pretty much says it all. If the Robot Controller app is opened when the Expansion Hub does not have 12v power, then a warning is added to the system telemetry explaining as much. However, if 12v is then restored, the warning is not removed until the user performs a "restart robot". Similarly, if the Robot Controller app is opened when the Expansion Hub *does* have 12v power, and then power is subsequently removed, the system telemetry warning is not issued until the user performs a "restart robot".

**Comment by FROGbots-4634** [comment]
Update: ok it seems like it is also refreshed when an OpMode is initialized, but it is the issue persists while no OpMode is running, as well as WHILE an OpMode is running.

**Comment by cmacfarl** [comment]
> but it is the issue persists while no OpMode is running

Well that makes sense as, technically speaking, there's is no time when an OpMode is not running.  It's either running a user OpMode, or it's running the builtin StopRobot OpMode.

**Comment by FROGbots-4634** [comment]
@cmacfarl  yeah I was trying to go for clarity but it seems that backfired. What I should have said is, "the issue persists WHILE an OpMode is running, whether it be a user OpMode or the DefaultOpMode."

**Comment by vwestin** [comment]
We found a very useful link to the battery voltage here:
https://www.reddit.com/r/FTC/comments/3odx26/is_it_possible_to_get_the_battery_voltage/
Which can be activated as:
        VoltageSensor   batteryVoltSensor;
        batteryVoltSensor = hardwareMap.voltageSensor.get("Expansion Hub 2");
        swerveDebugDevice(500,"Battery Voltage Sensor", batteryVoltSensor);
Teams can read the current voltage using:
        if (batteryVoltSensor.getVoltage() > 13.8)
            .....
This could also be added to the standard telemetry data that the team send to the driver station.

**Comment by AlecHub** [comment]
If you enumerate the voltage sensors, you don't have to hardcode device names, such as "Expansion Hub 2":
```java
List<VoltageSensor> voltageSensors;
VoltageSensor batteryVoltSensor = null;

voltageSensors = hardwareMap.getAll(VoltageSensor.class);

if (voltageSensors.size() > 0) {
     batteryVoltSensor = voltageSensors.get(0);
     for (VoltageSensor vs: voltageSensors) {
         telemetry.addData(vs.getDeviceName(), "Voltage: %.2f", vs.getVoltage());
     }
}
```
This works for both REV and MR battery voltage sensors.  **batteryVoltSensor** is then the first voltage sensor in the list, if any.

**Comment by NoahAndrews** [comment]
This issue is actually fixed in the 5.4 release: https://github.com/FIRST-Tech-Challenge/SkyStone/tree/v5.4

![image](https://user-images.githubusercontent.com/10224994/73012334-aabf0900-3ddb-11ea-9a7b-8e3c7b9f71e1.png)
